src/Tests/Microsoft.NET.Sdk.Web.Tests/PublishTests.cs

```diff
@@ -52,6 +52,28 @@ public void TrimmingOptions_Are_Defaulted_Correctly_On_Trimmed_Apps(string targe
                     .Should().BeFalse();
         }
 
+        [Fact]
+        public void TrimMode_Defaulted_Correctly_On_Trimmed_Apps_Pre_Net8()
+        {
+            var projectName = "HelloWorld";
+            var targetFramework = "net7.0";
+            var rid = EnvironmentInfo.GetCompatibleRid(targetFramework);
+
+            var testProject = CreateTestProjectForILLinkTesting(targetFramework, projectName);
+            testProject.AdditionalProperties["PublishTrimmed"] = "true";
+            testProject.AdditionalProperties["RuntimeIdentifier"] = rid;
+            testProject.SelfContained = "true";
+            testProject.PropertiesToRecord.Add("TrimMode");
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: projectName + targetFramework);
+
+            var publishCommand = new PublishCommand(testAsset);
+            publishCommand.Execute($"/p:RuntimeIdentifier={rid}").Should().Pass();
```

Comment:
We already set the RID on line 64.

```suggestion
            publishCommand.Execute().Should().Pass();
```